src/axolotl/utils/schemas/config.py

```
@@ -1114,19 +1121,92 @@ def check_qat_config(cls, data):
+    @classmethod
+    def check_fsdp_torch_version(cls, data):
+        env_capabilities = data.get("env_capabilities", {})
+        torch_version = env_capabilities.get("torch_version")
+
+        if torch_version is None:
+            import torch
+
+            torch_version = str(torch.__version__).split("+", maxsplit=1)[0]
+
+        if data.get("fsdp_config") and str(data.get("fsdp_version")) == "2":
             if version.parse(torch_version) < version.parse("2.7.0"):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)